### ISSUE TITLE
Add `goto_next_diag_error`/`goto_prev_diag_error`

### ISSUE THREAD

**Issue by AceofSpades5757** [helix-editor/helix#3405]
#### Describe your feature request

Wanted to continue the conversation on going to the next error. I found this _extremely_ useful in another editor and I get really annoyed because I usually have a lot of diagnostic points while I'm working out an issue.

[Original Issue](https://github.com/helix-editor/helix/issues/1929)

For reference: [Pull Request for next diagnostic](https://github.com/helix-editor/helix/pull/142/files)

**Comment by EpocSquadron** [comment]
This already exists for the current document with `]d` and `[d`. There is also document and workspace diagnostic pickers as of #2013

**Comment by the-mikedavis** [comment]
If I understand correctly, this is only about showing the next error diagnostic, right? `]d`/`[d` will bring you to the next diagnostic which could be info/warn/error

**Comment by EpocSquadron** [comment]
That makes a lot more sense now!

**Comment by AceofSpades5757** [comment]
Yeah, the diagnostics are great and all, but I'm usually trying to get to the next error first, then I can worry about the diagnostic details.

**Comment by 94bryanr** [comment]
I would also like to see this. Using the diagnostic picker in a workspace with hundreds/thousands of warnings requires a lot of scrolling around to find the next error.

I think having the diagnostics sorted by severity with errors at the top would make a lot of sense, as finding the next error (arguably the most common task you will be doing when opening the picker) would only require one extra keypress. Then within the severity buckets we could just continue to use the current filename sorting.

**Comment by cgahr** [comment]
There is also #6331, trying to implement the same thing.
